### PR TITLE
GetContainers() filter on metadata

### PR DESCRIPTION
**Pull request by ramo-j**
Added ability to set metadata on a container, and then use that metadata to filter retrieval of containers via state.GetContainer()

### CODE REVIEW

File: dftimewolf/lib/containers/interface.py [hunk 1]
```diff
@@ -50,3 +50,11 @@ def GetAttributeNames(self) -> List[str]:
       attribute_names.append(attribute_name)
 
     return attribute_names
+
+  def SetMetadata(self, key: str, value: Any) -> None:
+    """Sets metadata to the container.
+
+    Args:
+      key: Metadata key
+      value: Metadata value"""
```
Comment (jonathan-greig) [inline dftimewolf/lib/containers/interface.py:59]: Move closing quotes to next line.

File: dftimewolf/lib/state.py [hunk 2]
```diff
@@ -260,24 +260,47 @@ def GetStats(self) -> List[StatsEntry]:
...
+      metadata_filter_value (Optional[Any]): Metadata value to filter on.
 
     Returns:
       Collection[AttributeContainer]: attribute container objects provided in
           the store that correspond to the container type.
+
+    Raises:
+      RuntimeError: If only one metadata filter parameter is specified.
     """
+    if bool(metadata_filter_key) ^ bool(metadata_filter_value):  # logical XOR
```
Comment (jonathan-greig) [inline dftimewolf/lib/state.py:282]: Maybe find a more Pythonic way of checking this, something like `if None in (metadata_filter_key, metadata_filter_value).

**Review by jonathan-greig — DISMISSED** [review]
A few nits but otherwise looks good.

File: dftimewolf/lib/state.py (hunk 2 above)
Comment (jonathan-greig) [reply dftimewolf/lib/state.py:282]: Disregard this suggestion, I misunderstood what you're trying to do with this line.

File: dftimewolf/lib/containers/interface.py (hunk 1 above)
Comment (ramo-j) [reply dftimewolf/lib/containers/interface.py]: Done.

**Review by jonathan-greig — APPROVED** [review]

**Pull request merged**